### DISCUSSION TITLE
disconnected: ** (DBConnection.ConnectionError) client #PID<[XXX]> exited

### DISCUSSION THREAD

**Discussion by jonleighton** [keathley/wallaby#616] [General]
I thought I'd share that I've been working recently on fixing a flakey Wallaby test that was sometimes failing on CI with the above error.

I managed to isolate the cause, which is explained here: https://github.com/elixir-ecto/db_connection/issues/247

TL;DR: when a browser aborts a request, the Cowboy2 server kills the process tree associated with that request. If a database connection (shared via `Phoenix.Ecto.SQL.Sandbox`) was in use at that time, the connection would become corrupted. After that, future requests, and the test that owns the connection, would be unable to use it.

Here's a test which reproduces the situation: https://github.com/jonleighton/db_connection_ownership_bug/commit/922bbd8f4d3d6c64780ccc9385bf52b91fa9ce2d

And a fix has already gone into `phoenix_ecto`, so this shouldn't happen in future releases: https://github.com/phoenixframework/phoenix_ecto/commit/1d8d28a82eef1ff496851bd2fd53661398d38b99

**Comment by mhanberg** [comment]
Thanks!
